egg/zoo/dsprites_bvae/train.py

```diff
@@ -0,0 +1,210 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source python -m egg.zoo.dsprites_bvae.train --lr=1e-3 --batch_size=128 --n_epochs=100 --vocab_size=2tree.
+
+import torch
+import torch.utils.data
+from scipy.stats import spearmanr
+from torchvision import datasets, transforms, utils
+from torch import nn
+from torch.nn import functional as F
+import egg.core as core
+import pathlib
+from torch.autograd import Variable
+import json
+from typing import Union, Callable
+import os
+
+
+
+import numpy as np
+from .archs import VisualReceiver, VisualSender
+from egg.zoo.dsprites_bvae.data_loaders.data_loaders import get_dsprites_dataloader
+from egg.core.language_analysis import TopographicSimilarity, PosDisent
+
+
+class TopographicSimilarityLatents(TopographicSimilarity):
+    def __init__(self,
+                 sender_input_distance_fn: Union[str, Callable] = 'cosine',
+                 message_distance_fn: Union[str, Callable] = 'edit',
+                 compute_topsim_train_set: bool = False,
+                 compute_topsim_test_set: bool = True):
+
+        super().__init__(sender_input_distance_fn, message_distance_fn, compute_topsim_train_set, compute_topsim_test_set)
+
+    def compute_similarity(self, sender_input: torch.Tensor, messages: torch.Tensor, mode: str,  epoch: int):
+        def compute_distance(_list, distance):
+            return [distance(el1, el2)
+                    for i, el1 in enumerate(_list[:-1])
+                    for j, el2 in enumerate(_list[i+1:])
+                    ]
+
+        messages = [msg.tolist() for msg in messages]
+        input_dist = compute_distance(
+            sender_input.numpy(), self.sender_input_distance_fn)
+        message_dist = compute_distance(messages, self.message_distance_fn)
+        topsim = spearmanr(input_dist, message_dist,
+                           nan_policy='raise').correlation
+
+        output_message = json.dumps(dict(topsim=topsim, epoch=epoch))
+        print(output_message, flush=True)
```

Comment:
Are there any difference with the egg implementation except for the argmax on the messages?

Would it be the same if you create the callback with TopographicSimilarity(sender_input_distance_fn='euclidean', message_distance_fn='euclidean', is_gumbel=False)?

Reply:
Yes, good point! I removed the child class